Mage/src/mage/filter/predicate/mageobject/AbilityPredicate.java

```diff
@@ -57,7 +60,19 @@ public boolean apply(MageObject input, Game game) {
         
         for (Ability ability : abilities) {
             if (abilityClass.equals(ability.getClass())) {
-                return true;
+                //Burseg Hack: Magewright's Stone [DIS] specifically looks for activated ability containing {T}
+                //As of 9/13/2015, no other AbilityPredicate is filtering for ActivatedAbilityImpl 
+                //if (abilityClass.equals(SimpleActivatedAbility.class) && ability.getCosts().size() > 0){
+                if((ability instanceof ActivatedAbilityImpl) && ability.getCosts().size() > 0){
+                    for (Cost cost : ability.getCosts()) {
+                        if (cost instanceof TapSourceCost) {
+                            return true;
+                        }
+                    }
+                    return false;
+                } else {
+                    return true;
+                }

```

Comment:
Ok, I'll get to work on these changes when I get home from work, thanks for the feedback.